Setup/UpgradeSchema.php

```diff
@@ -291,4 +295,198 @@ protected function version_2_4_9(ModuleContextInterface $context, SchemaSetupInt
+                'nullable' => false,
+                'default' => Table::TIMESTAMP_INIT
+            ], 'When create')
+            ->addIndex(
+                $setup->getIdxName('tnw_salesforce_entity_queue', ['code', 'entity_id', 'entity_load']),
+                ['code', 'entity_id', 'entity_load']
+            )
+            ->addIndex(
+                $setup->getIdxName('tnw_salesforce_entity_queue', ['transaction_uid', 'code', 'status', 'website_id']),
+                ['transaction_uid', 'code', 'status', 'website_id']
+            )
+            ->addForeignKey(
+                $setup->getFkName('tnw_salesforce_entity_queue', 'website_id', 'store_website', 'website_id'),
+                'website_id',
+                $setup->getTable('store_website'),
+                'website_id',
+                Table::ACTION_CASCADE
+            )
+        ;
+
+        $setup->getConnection()
+            ->createTable($table);
+
+        $table = $setup->getConnection()
+            ->newTable($setup->getTable('tnw_salesforce_entity_queue_relation'))
+            ->addColumn('queue_id', Table::TYPE_INTEGER, null, [
+                'unsigned' => true,
+                'nullable' => false,
+            ], 'Queue Id')
+            ->addColumn('parent_id', Table::TYPE_INTEGER, null, [
+                'unsigned' => true,
+                'nullable' => false,
+            ], 'Parent Id')
+            ->addIndex(
+                $setup->getIdxName(
+                    'tnw_salesforce_entity_queue_relation',
+                    ['queue_id', 'parent_id'],
+                    \Magento\Framework\DB\Adapter\AdapterInterface::INDEX_TYPE_UNIQUE
+                ),
+                ['queue_id', 'parent_id'],
+                ['type' => \Magento\Framework\DB\Adapter\AdapterInterface::INDEX_TYPE_UNIQUE]
+            )
+            ->addForeignKey(
+                $setup->getFkName(
+                    'tnw_salesforce_entity_queue_relation',
+                    'queue_id',
+                    'tnw_salesforce_entity_queue',
+                    'queue_id'
+                ),
+                'queue_id',
+                $setup->getTable('tnw_salesforce_entity_queue'),
+                'queue_id',
+                Table::ACTION_CASCADE
+            )
+            ->addForeignKey(
+                $setup->getFkName(
+                    'tnw_salesforce_entity_queue_relation',
+                    'parent_id',
+                    'tnw_salesforce_entity_queue',
+                    'queue_id'
+                ),
+                'parent_id',
+                $setup->getTable('tnw_salesforce_entity_queue'),
+                'queue_id',
+                Table::ACTION_CASCADE
+            )
+        ;
+
+        $setup->getConnection()
+            ->createTable($table);
+    }
+
+    protected function version_2_4_14(ModuleContextInterface $context, SchemaSetupInterface $setup)
+    {
+        if (version_compare($context->getVersion(), '2.4.14') >= 0) {
```

Comment:
We need to apply it in version 2.4.24 (or 2.5.1): we already released v. 2.4.23 and installed for our clients.